avocado/utils/dmesg.py

```diff
@@ -0,0 +1,115 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2020 IBM
+# Author: Praveen K Pandey <praveen@linux.vnet.ibm.com>
+#
+
+"""
+Module for manipulate dmesg while running test.
+"""
+
+
+import time
+import os
+import tempfile
+import logging
+
+from . import process, genio
+
+LOGGER = logging.getLogger('avocado.test')
+
+
+class DmesgError(Exception):
+    """Base Exception Class for all dmesg  utils exceptions"""
+    pass
+
+
+def clear_dmesg():
+    """This function clear dmesg of system and save older dmesg in a file"""
+
+    cmd = "dmesg -c"
+    status = process.system(cmd, timeout=30, ignore_status=True,
+                            verbose=False, shell=True, sudo=True)
+    if status:
+        raise DmesgError(
+            "Unable to clear dmesg as some issue while clearing")
+
+
+def collect_dmesg(output_file=None):
+    """This Function collect dmesg and save in a tmp file
+
+    :return: file which contain dmesg
+    """
+    if output_file is None:
+        fd, output_file = tempfile.mkstemp(
+            suffix=".-%s" % time.strftime("%Y-%m-%d:%H:%M:%S"), dir=tempfile.gettempdir())
+    dmesg = process.system_output(
+        "dmesg", ignore_status=True, sudo=True).decode()
+    genio.write_file(output_file, dmesg)
+    if not os.path.isfile(output_file):
+        raise DmesgError("{} is not a valid file.".format(output_file))
+    return output_file
+
+
+def verify_patterns_dmesg(dmesg_log_file=None, patterns=None):
+    """check patterns in dmesg
+
+    :output_file: Used for process and find patterns
+                     If None use internal collect_dmesg
+                     get tmp demsg file
+    :patterns      :  List variable to search in dmesg
+                     If None  assign  default patterns to "Call Trace:"
+    :return error log in form of list
+    """
+    error = []
+    dmesg_log_file = collect_dmesg(dmesg_log_file)
+    if not patterns:
+        patterns = ["Call Trace:"]
+    for fail_pattern in patterns:
+        for log in genio.read_file(dmesg_log_file).splitlines():
+            if fail_pattern in log:
+                error.append(log)
+    return error
+
+
+def verify_dmesg(dmesg_log_file=None, level_check=5, ignore_result=False):
+    """verify dmesg as find Call traces in dmesg log using severity
```

Comment:
s/^v/V/g
s/$/./g

Reply:
By looking to this function name and docstring, it is not clear to me what this is making. why Call traces?

Reply:
ah let me think what can be best way